src/redux/ui/screens/data/actions.ts

```diff
@@ -0,0 +1,83 @@
+import { Dispatch } from 'react';
+import { IReduxState } from '../../../state';
+import {
+    IScreenContext,
+    IScreenRequest,
+    IScreenResponse
+} from '../../../../components/widgets/types';
+import { ApiClient } from '../../../../core/utils/api-client/api-client';
+import { AccountType } from '../../../wallets/state';
+import { IAction } from '../../../types';
+import { getSelectedAccount, getSelectedWallet } from '../../../wallets/selectors';
+import { Platform } from 'react-native';
+import { getChainId } from '../../../preferences/selectors';
+
+export const FETCH_SCREEN_DATA = 'FETCH_SCREEN_DATA';
+
+export const fetchScreenData = (context: IScreenContext) => async (
+    dispatch: Dispatch<
+        IAction<{
+            request: IScreenRequest;
+            response: IScreenResponse;
+            isLoading: boolean;
+            error: any;
+        }>
+    >,
+    getState: () => IReduxState
+) => {
+    const state = getState();
+    const wallet = getSelectedWallet(state);
+    const account = getSelectedAccount(state);
+    const chainId = getChainId(state, account.blockchain);
+
+    const apiClient = new ApiClient();
+
+    const body: IScreenRequest = {
+        context: {
+            screen: context.screen,
+            tab: context?.tab
+        },
+        user: {
+            os: Platform.OS as 'ios' | 'android',
+            deviceId: state.preferences.deviceId,
+            theme: 'dark',
+            country: 'Moon',
+            lang: 'en',
+
+            wallet: {
+                pubKey: wallet.walletPublicKey,
+                type: wallet.type
+            },
+
+            blockchain: account.blockchain,
+            chainId: String(chainId),
+            address: account.address,
+            accountType: AccountType.DEFAULT
+        }
+    };
+
+    try {
+        const screenResponse = await apiClient.http.post('/walletUi/screen', body);
```

Comment:
you should update the state before doing the request, switch `isLoading` to `true` and assign `undefined` to `error` field